### ISSUE TITLE
RetryPolicy creation is order dependent

### ISSUE THREAD

**Issue by nheitz** [michaelbull/kotlin-retry#2]
Thanks for this lib!  I've had one minor hiccup whilst utilising in my current code, as demonstrated by this simple unit test:

```
package somepackage

import com.github.michaelbull.retry.ContinueRetrying
import com.github.michaelbull.retry.StopRetrying
import com.github.michaelbull.retry.policy.RetryPolicy
import com.github.michaelbull.retry.policy.fullJitterBackoff
import com.github.michaelbull.retry.policy.limitAttempts
import com.github.michaelbull.retry.policy.plus
import com.github.michaelbull.retry.retry
import kotlinx.coroutines.runBlocking
import org.junit.Test

class RetryTest {
    private val retryFailures: RetryPolicy<Throwable> = {
        when (reason) {
            is RuntimeException -> ContinueRetrying

            else -> StopRetrying
        }
    }

    @Test
    fun doesnotwork() = runBlocking {
        // this fails because the RetryPolicy is considered incorrect...delayMillis equals 0 after first '+'         
        retry(retryFailures + limitAttempts(5) + fullJitterBackoff(100, 1000)) {
            doSomethingWrong()
        }
    }

    @Test
    fun works() = runBlocking {
        retry(fullJitterBackoff(100, 1000) + retryFailures + limitAttempts(5)) {
            doSomethingWrong()
        }
    }


    private fun doSomethingWrong() {
        throw RuntimeException("Gotcha")
    }
}
```

It seems that the plus operator is prematurely enforcing the validation of the final Policy before all elements might have been added in.  Perhaps the 'delayMillis' validation can be done after this line in Retry.kt?
```
                val instruction = RetryFailure(t).policy()
```

**Comment by michaelbull** [comment]
Thanks very much for the bug report and for the detailed explanation. I'll have a look over the implementation this weekend.

>It seems that the plus operator is prematurely enforcing the validation of the final Policy before all elements might have been added in.

I'm not sure this is the case. When you 'combine' two policies, you're basically making a new policy that, when evaluated, will return the higher delay of the two. Also, the concept of a zero-value `delayMillis` is indicative of the `ContinueRetrying` instruction, so I'm not sure there's any validation to be done inside `Retry.kt`.

**Comment by michaelbull** [comment]
This should be fixed in 4e1cf758c0799801398c09142ca53185a2cff537, which I'll endeavour to push a new release for this evening. Thanks for the bug report!
